Description: Don't create native file type associations automatically.
Author: Jens Reyer <jre.winesim@gmail.com>
Bug: https://bugs.winehq.org/show_bug.cgi?id=28159
Bug-Debian: https://bugs.debian.org/845334

--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -3309,7 +3309,7 @@ HKR,Parameters,"ServiceDll",,"%11%\sched
 HKLM,%CurrentVersionNT%\SvcHost,"netsvcs",0x00010008,"Schedule"
 
 [Services]
-HKLM,%CurrentVersion%\RunServices,"winemenubuilder",2,"%11%\winemenubuilder.exe -a -r"
+HKLM,%CurrentVersion%\RunServices,"winemenubuilder",2,"%11%\winemenubuilder.exe -r"
 HKLM,"System\CurrentControlSet\Services\Eventlog\Application",,16
 HKLM,"System\CurrentControlSet\Services\Eventlog\System",,16
 HKLM,"System\CurrentControlSet\Services\Tcpip\Parameters",,16
--- a/dlls/shell32/changenotify.c
+++ b/dlls/shell32/changenotify.c
@@ -415,9 +415,9 @@ void WINAPI SHChangeNotify(LONG wEventId
 
     if (wEventId & SHCNE_ASSOCCHANGED)
     {
-        static const WCHAR args[] = {' ','-','a',0 };
         TRACE("refreshing file type associations\n");
-        run_winemenubuilder( args );
+        TRACE("refreshing native associations is disabled deliberately in Debian\n");
+        run_winemenubuilder;
     }
 }
 
